homeassistant/components/starlink/__init__.py

```diff
@@ -0,0 +1,35 @@
+"""The Starlink integration."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_IP_ADDRESS, Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .coordinator import StarlinkUpdateCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Starlink from a config entry."""
+    coordinator = StarlinkUpdateCoordinator(
+        hass=hass,
+        url=entry.data[CONF_IP_ADDRESS],
+        name=entry.title,
+    )
+
+    await coordinator.async_config_entry_first_refresh()
+
+    hass.data[DOMAIN] = {entry.entry_id: coordinator}
```

Comment:
If they have multiple starlink integration this will override the data for the domain. use `setdefault` instead